lib/screens/config/state.ex

```diff
@@ -169,20 +174,24 @@ defmodule Screens.Config.State do
     {:reply, app_params, state}
   end
 
-  def handle_call({:mode_disabled?, mode}, _from, %__MODULE__{config: config} = state) do
-    %Devops{disabled_modes: disabled_modes} = config.devops
+  def handle_call({:screen_ids, filter_fn}, _from, %__MODULE__{config: config} = state)
+      when is_function(filter_fn, 1) do
```

Comment:
Nit: since this guard is already applied in the "client" function above, it seems somewhat redundant to have it here too. What do you think of matching explicitly against the default/`nil` case first, and then having this pattern defined below without the guard?

Reply:
Oops good catch! Yes, the client function should be responsible for guarding against bad input, not the server handler function.
I think I did this to separate this case from the one where `filter_fn` is `nil`, but I could just swap the two clauses and it would work fine without a guard.